csp_billing_adapter/adapter.py

```diff
@@ -128,27 +142,56 @@ def event_loop_handler(
     """Perform the event loop processing actions."""
     log.info('Starting event loop processing')
 
-    usage = hook.get_usage_data(config=config)
-    log.debug('Retrieved usage data: %s', usage)
-
-    add_usage_record(hook, config, usage)
-
-    cache = hook.get_cache(config=config)
-    now = get_now()
-
-    log.debug(
-        "Now: %s, Next Reporting Time: %s, Next Bill Time: %s",
-        date_to_string(now),
-        cache['next_reporting_time'],
-        cache['next_bill_time']
-    )
+    try:
+        usage = hook.get_usage_data(config=config)
+    except Exception as exc:
+        log.warning("Failed to retrieve usage data: %s", str(exc))
+        now = get_now()
+        hook.update_csp_config(
+            config=config,
+            csp_config={
+                'timestamp': date_to_string(now),
+                'billing_api_access_ok': False,
+                'errors': [
+                    f'Usage data retrieval failed: {exc}'
```

Comment:
It might be helpful to track state during each cycle and any error messages and update it at the end of the loop. This way the state of the adapter is updated separately of the actions and state independent data.

Reply:
Agreed, but for such a case we may want to clarify how that is reflected in the csp_config data, as per my comments in #70.